### PR TITLE
Update elastic_agent.py. Delete _get_socket_with_port import

### PR DESCRIPTION
**Pull request by QiaoZhennn**
cannot import _get_socket_with_port from torch.distributed.elastic.agent.server.api

### CODE REVIEW

File: deepspeed/elasticity/elastic_agent.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from torch.distributed.elastic.agent.server.local_elastic_agent import LocalElasticAgent
 from typing import Any, Dict, Optional, Tuple
 from datetime import datetime
-from torch.distributed.elastic.agent.server.api import _get_socket_with_port
+from torch.distributed.elastic.utils.distributed import get_free_port as _get_free_port
```
Comment (kit1980) [inline deepspeed/elasticity/elastic_agent.py:9]: Just use `get_free_port` at the calling site instead of importing as old private name?

**Comment by kit1980** [comment]
Context: `_get_socket_with_port` has been removed from PyTorch in https://github.com/pytorch/pytorch/pull/125743 (and it was a private API, so no backward compatibility guarantee).

**Comment by adk9** [comment]
Thanks for the fix. This is a duplicate of #5654.

**Comment by loadams** [comment]
Closing as a duplicate of #5654 as that PR is set to be merged.

**Pull request closed without merging**